drivers/gnss/gnss_parse.c

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2023 Trackunit Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+
+#include <string.h>
+#include <errno.h>
+#include <stdlib.h>
+
+#include "gnss_parse.h"
+
+#define GNSS_PARSE_NANO_KNOTS_IN_MMS               (1943840LL)
+#define GNSS_PARSE_NANO                            (1000000000LL)
+#define GNSS_PARSE_MICRO                           (1000000LL)
+#define GNSS_PARSE_MILLI                           (1000LL)
+
+int gnss_parse_dec_to_nano(const char *str, int64_t *nano)
+{
+	int64_t sum = 0;
+	int8_t decimal = -1;
+	int8_t pos = 0;
+	int8_t start = 0;
+	int64_t increment;
+
+	__ASSERT(str != NULL, "str argument must be provided");
+	__ASSERT(str != NULL, "nano argument must be provided");
```

Comment:
str is being checked instead of nano

Reply:
could you send a PR with the fix?

Reply:
This is the [contribution guideline documentation](https://docs.zephyrproject.org/latest/contribute/guidelines.html#contribution-guidelines).